homeassistant/components/radarr/calendar.py

```diff
@@ -1,119 +1,55 @@
 """Support for Radarr calendar items."""
 from __future__ import annotations
 
-import asyncio
-from dataclasses import dataclass
-from datetime import date, datetime, timedelta
-
-from aiopyarr import RadarrCalendarItem
+from datetime import datetime
 
 from homeassistant.components.calendar import CalendarEntity, CalendarEvent
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityDescription
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
-from homeassistant.util import Throttle
 
 from . import RadarrEntity
 from .const import DOMAIN
-from .coordinator import RadarrDataUpdateCoordinator
-
-MIN_TIME_BETWEEN_UPDATES = timedelta(hours=1)
-
-
-@dataclass
-class RadarrEventMixIn:
-    """Mixin for Radarr calendar event."""
-
-    release_type: str
-
-
-@dataclass
-class RadarrEvent(CalendarEvent, RadarrEventMixIn):
-    """A class to describe a Radarr calendar event."""
+from .coordinator import CalendarUpdateCoordinator, RadarrEvent
 
-    description: str
+CALENDAR_TYPE = EntityDescription(
+    key="calendar",
+    name=None,
+)
 
 
 async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Set up the Radarr calendar entity."""
-    coordinator = hass.data[DOMAIN][entry.entry_id]["disk_space"]
-    async_add_entities([RadarrCalendarEntity(coordinator)], True)
+    coordinator = hass.data[DOMAIN][entry.entry_id]["calendar"]
+    async_add_entities([RadarrCalendarEntity(coordinator, CALENDAR_TYPE)])
 
 
 class RadarrCalendarEntity(RadarrEntity, CalendarEntity):
     """A Radarr calendar entity."""
 
-    def __init__(self, coordinator: RadarrDataUpdateCoordinator) -> None:
-        """Create the Calendar event device."""
-        super().__init__(coordinator, name=DOMAIN)
-        self.coordinator = coordinator
-        self._event: RadarrEvent | None = None
-        self._events: list[RadarrEvent] = []
+    coordinator: CalendarUpdateCoordinator
 
     @property
     def event(self) -> CalendarEvent | None:
         """Return the next upcoming event."""
-        return self._event
+        if not self.coordinator.event:
+            return None
+        self._attr_extra_state_attributes = {
```

Comment:
It seems a little surprising to mutate the properties of an object in a property method.  Perhaps it needs to be updated once when the coordinator callback fires?


(Aside: since we're talking about state updates -- calendar is now fixed to no longer need a separate poll interval for data fetching and a separate interval for state updates as events start. this is now handled by the base class. i don't think this needs to change anything, but just wanted to mention that if you hadn't seen it)